### PR TITLE
Publish to Sonatype

### PR DESCRIPTION
**Pull request by axuan25**
- Revised build and module structure for simplification
- Moved demo app outside of repository (having both `com.android.library` and `com.android.application` plugins was making life difficult)
- Defined maven publications for Sonatype
- Published and released `6.2.1` to ensure demo app can successfully pull dependency with no breaking changes (✅ )

Next:
- Revise package FQDNs (breaking)
- Upgrade Gradle
- Upgrade Kotlin
- Upgrade deps
- Migrate to Kotlin syntax for build ?

### CODE REVIEW

**Review by BarPierre — APPROVED** [review]

**Review by StanWarz — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -1,39 +1,147 @@
...
-}
 
-task clean(type: Delete) {
-    delete rootProject.buildDir
+    afterEvaluate {
+        publishing {
+            repositories {
+                maven {
+                    name = "release"
+                    url = "https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/"
```
Comment (thomasreach5) [inline build.gradle:88]: The URL looks like it points to a specific host.

File: sdk-core/build.gradle [hunk 2]
```diff
@@ -2,51 +2,15 @@ apply plugin: "com.android.library"
...
-    implementation "com.android.support:customtabs:28.0.0"
-    implementation "com.github.kittinunf.result:result:2.2.0"
-    implementation "com.google.android.gms:play-services-fido:18.1.0"
+    implementation ext.kotlin_stdlib
+    implementation libs.androidCustomTabs
+    implementation libs.result
+    implementation libs.playServicesFido
+    implementation libs.retrofit
+    implementation libs.retrofitConverterGson
     implementation "com.squareup.okhttp3:logging-interceptor:4.0.0"
```
Comment (thomasreach5) [inline sdk-core/build.gradle:15]: Can this also be moved to `ext.libraries` like the others ?

**Review by thomasreach5 — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (axuan25) [reply build.gradle:88]: Yes, this is the new URL for Sonatype. `https://oss.sonatype.org` is now legacy.

File: sdk-core/build.gradle (hunk 2 above)
Comment (axuan25) [reply sdk-core/build.gradle:15]: This library is only used within this module, so I preferred to not include it there.

**Pull request merged**